shared/ssz/example_encode_test.go

```diff
@@ -0,0 +1,36 @@
+package ssz_test
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/prysmaticlabs/prysm/shared/ssz"
+)
+
+// SSZ encoding takes a given data object to the given io.Writer. The most
+// common use case is to use bytes.Buffer to collect the results to a buffer
+// and consume the result.
+func ExampleEncode() {
+	// Given a data structure like this.
+	type data struct {
+		Field1 uint8
+		Field2 []byte
+	}
+
+	// And some basic data.
+	d := data{
+		Field1: 10,
+		Field2: []byte{1, 2, 3, 4},
+	}
+
+	// We use a bytes.Buffer as our io.Writer.
+	buffer := new(bytes.Buffer)
+	// ssz.Encode writes the encoded data to the buffer.
+	if err := ssz.Encode(buffer, d); err != nil {
+		// There was some failure with encoding SSZ.
+		panic(err)
```

Comment:
Should it be a panic ? A normal error would be fine here

Reply:
It's just for example, see below

![image](https://user-images.githubusercontent.com/7246818/53259791-fc2bb900-369d-11e9-978b-a6f54edd3537.png)

Reply:
I'll add another comment that the user should handle the error rather than panic.

Reply:
Ah k got it. Sounds good